From: Tomasz Majchrzak <tomasz.majchrzak@intel.com>
Date: Thu, 30 Jun 2016 10:47:09 +0200
Subject: md: add missing sysfs_notify on array_state update
References: bsc#1003941,FATE#321732
Git-commit: 573275b58ee9e1d2ec89f9608060f58931c3cde3
Patch-Mainline: v4.8-rc1

Changeset 6791875e2e53 has added early return from a function so there is no
sysfs notification for 'active' and 'clean' state change.

Signed-off-by: Tomasz Majchrzak <tomasz.majchrzak@intel.com>
Signed-off-by: Shaohua Li <shli@fb.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/md.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 20ff8d6..effd4ef 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -3944,6 +3944,8 @@ array_state_store(struct mddev *mddev, const char *buf, size_t len)
 			} else
 				err = -EBUSY;
 		}
+		if (!err)
+			sysfs_notify_dirent_safe(mddev->sysfs_state);
 		spin_unlock(&mddev->lock);
 		return err ?: len;
 	}
-- 
1.8.5.6

